src/tools/tor-print-ed-signing-cert.c

```diff
@@ -55,9 +56,26 @@ main(int argc, char **argv)
     return -5;
   }
 
-  time_t expires_at = (time_t)cert->exp_field * 60 * 60;
+  const time_t expiration = (const time_t)cert->exp_field * 60 * 60;
 
-  printf("Expires at: %s", ctime(&expires_at));
+  const struct tm *expires_at = localtime(&expiration);
+
+  char rfc822_str[16] = "";
```

Comment:
Looks like there's some kind of memory safety issue here, I suggest using 64 characters for RFC 822. (It's about 32, but let's not mess around.)